metric-example/metric-expressions-by-example.md

```diff
@@ -0,0 +1,248 @@
+* operands of arithmetic are written in parentheses,
+* `/` and `*` bind more closely than `-` and `+`, but just like in math, we can define a different order of operations by enclosing sub-expressions in parentheses,
+* operators like `:filter` can be applied to a bracketed sub-expression with a calculation in it.
+
+Keep in mind that many metric keys have dashes in them. Those dashes are interpreted literally (as part of the metric key)
+and only become a calculation once the operands are enclosed in parentheses. For example, `request-duration` is a single
+metric key, while `(request)-(duration)` subtracts `duration` from `request`.
+
+## Scenario 2: Percentage of Client Errors
+> **Task** We recently made a breaking change to our API and want to see which percentage
+> of calls to the REST service trigger 4xx errors before and after the introduction of the
+> breaking change.
+
+`builtin:service.errors.fourxx.count` gives us just the amount of service calls with a 4xx exit
+status, while `builtin:service.errors.total.count` additionally includes 5xx errors. If our
+REST service has the entity ID `SERVICE-1234567890`, then we can get the percentage of errors
+in the 4xx range with:
+```
+(
+  (100)
+  *
+  (builtin:service.errors.fourxx.count)
+  /
+  (builtin:service.errors.total.count)
+):filter(eq("dt.entity.service","SERVICE-1234567890"))
+```
+
+How does such a calculation with two separate metrics work? Dynatrace will first find all the series
+with equal dimensions. For each match, we combine the associated data at the same time slot by applying
+the numbers to the arithmetic operator in use. With the result, we assemble a new output series for each
+match.
+
+In the case of our example, we use this functionality to divide error counts that were measured in the
+same time slot.
+
+## Scenario 3: Revenue in Percent over the Query Timeframe
+> **Task** Which percentage of our revenue was made in any given month of the last year?
+
+We store revenue data in a metric called `revenue` that is split by a single dimension
+`region`. Instead of revenue in absolute terms, we want to see for each timeslot how many
+percent of the revenue over the whole query timeframe fall into that timeslot. In other
+words, we want to divide by the combined revenue over the whole query timeframe and multiply
+by 100.
+
+For the multiplication by `100`, we can just a numeric literal, but for the combined
+revenue we would need a calculated value instead of a fixed one.
+
+To achieve that with metric expressions, we need to reduce the revenue result to a
+single value. This is done by first merging all dimensions into one using `splitBy()`,
+yielding a single dimension-less series. Next, we want to apply `fold` to sum up all of
+the individual data points of this one series, so that we end up with a single number
+for the full revenue.
+
+Putting it all together, we set our resolution to months, set `from` to `now-1y to now`
```

Comment:
Is that right? `from` to `now-1y to now`
Doesn't `from` just become `now-1y`?

Reply:
Both are correct, we can:
* leave out `to` (and rely on the default value `now`) and set `from` to `now-1y` as you suggested,
* set `from` to `now-1y to now`,
* set `from` to `now-1y` and `to` to `now`.

Do you want to leave it as-is or should we go with the shorter version?

Reply:
As long as it's not a mistake we're good 👍